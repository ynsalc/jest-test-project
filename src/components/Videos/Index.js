import React from "react";

const Index = ({ videos = [] }) => {
  if (videos.length <= 0) {
    return <h2>Sorry, no videos!</h2>;
  }
  return (
    <div>
      {videos.map((video, index) => (
        <h3 key={index}>{video}</h3>
      ))}
    </div>
  );
};

export default Index;
