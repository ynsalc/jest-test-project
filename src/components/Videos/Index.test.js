import React from "react";
import renderer from "react-test-renderer";
import Videos from './Index';

describe('Videos component unit test', () => {
    test('Videos empty renders sorry, no videos found', () => {
        const tree = renderer.create(<Videos />).toJSON();

        expect(tree).toMatchSnapshot();
    })

    test('Videos 1 item renders correctly', () => {
        const tree = renderer.create(<Videos videos={["udemy"]} />).toJSON();

        expect(tree).toMatchSnapshot();
    })
})