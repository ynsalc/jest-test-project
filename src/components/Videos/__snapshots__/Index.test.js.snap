// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Videos component unit test Videos 1 item renders correctly 1`] = `
<div>
  <h3>
    udemy
  </h3>
</div>
`;

exports[`Videos component unit test Videos empty renders sorry, no videos found 1`] = `
<h2>
  Sorry, no videos!
</h2>
`;
